### ISSUE TITLE
Crash with the latest version

### ISSUE THREAD

**Issue by JomerDev** [RSDuck/RandomPeripherals#11]
My MC just crashed with the lates version of the mod:

Time: 07.05.16 18:47
Description: Unexpected error

java.lang.NullPointerException: Unexpected error
    at me.kemal.randomp.client.IconGeneratorScreen.<init>(IconGeneratorScreen.java:47)
    at me.kemal.randomp.client.ClientProxy.generateItem3DIcons(ClientProxy.java:94)
    at cpw.mods.fml.common.eventhandler.ASMEventHandler_1521_ClientProxy_generateItem3DIcons_Load.invoke(.dynamic)
    at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54)
    at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140)
    at net.minecraft.client.multiplayer.WorldClient.<init>(WorldClient.java:61)
    at com.rwtema.funkylocomotion.fakes.FakeWorldClient.<init>(FakeWorldClient.java:42)
    at com.rwtema.funkylocomotion.fakes.FakeWorldClient.getFakeWorldWrapper(FakeWorldClient.java:54)
    at com.rwtema.funkylocomotion.rendering.TileEntityRenderMoving.createCache(TileEntityRenderMoving.java:250)
    at com.rwtema.funkylocomotion.rendering.TileEntityRenderMoving.func_147496_a(TileEntityRenderMoving.java:241)
    at net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher.func_147543_a(SourceFile:117)
    at net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher.func_147542_a(SourceFile:71)
    at fastcraft.ak.a(F:450)
    at fastcraft.HC.al(F:50)
    at net.minecraft.client.renderer.RenderGlobal.func_147589_a(RenderGlobal.java)
    at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1337)
    at net.minecraft.client.renderer.EntityRenderer.func_78480_b(EntityRenderer.java:1011)
    at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1001)
    at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:898)
    at net.minecraft.client.main.Main.main(SourceFile:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

**Comment by RSDuck** [comment]
I think this should do it:
[RandomPeripherals-1.5 Beta 2.jar.zip](https://github.com/RSDuck/RandomPeripherals/files/253571/RandomPeripherals-1.5.Beta.2.jar.zip)

**Comment by Nathan1852** [comment]
It may. (The crash only appeared once) But I have another problem, which also happened with the first 1.5. When I try to start or load a world, the icons will all appear (which I assume is normal) but when it is finished with loading them it will then only render the normal ingame UI with a completely black background (no blocks in sight). It the starts to switch every ~2 seconds between the UI beeing visible and it beeing not visible. When I try to move the mouse or click the game will freeze for a few seconds and then it ill display the 'Disconnected from Server' screen, despite me never joining a server but only loading a local game

**Comment by RSDuck** [comment]
Could you please give me more informations about what mods you have installed?

I tried to reproduce your crash by adding funcy locomotion, but in my case it worked without problem after renaming the beta 2 file from .jar.zip to .jar(Github doesn't allows .jar files here)

**Comment by Nathan1852** [comment]
This is the complete log from where I had the problem: http://puu.sh/oJMIx/349c62d3b8.log
If there is any data missing that you need, message me.

**Comment by RSDuck** [comment]
I found the problem(whoo the log is long). It isn't even related to the rendering of the blocks. Instead the problem is that there are so many items that the size of the rendered images is so big that they can't be sended in one packet from client to server

**Comment by RSDuck** [comment]
I found the problem(whoo the log is long). It isn't even related to the rendering of the blocks. Instead the problem is that there are so many items that the size of the rendered images is so big that they can't be sended in one packet from client to server.

Try this file(and don't forget to rename it to .jar otherwise another error occurs), I modified it so that all images get sended in seperate packets:

[RandomPeripherals-1.5 Beta 2.jar.zip](https://github.com/RSDuck/RandomPeripherals/files/253879/RandomPeripherals-1.5.Beta.2.jar.zip)
